azerothcore/azerothcore-wotlk
# fix(Scripts/HoS): Adjust IsEncounterInProgress()

## Changes Proposed:
-  fix `IsEncounterInProgress()` in `instance_halls_of_stone.cpp`, which returned `true` when Brann's data changed (it seems like it's used to keep track of the escort state, the bosses' encounters states are properly stored in other indices in the `Encounter` array).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5497

## Tests Performed:

- Applied the C++ fix
- Tested on Windows 10 client and server


## How to Test the Changes:

1. Get two characters into a group
2. `.tele hallsofstone`, enter the instance
3. Do the dungeon, from time to time, try to have one character die and revive it with the other one (with a proper spell, not with `.revive`)
4. Make sure that if you fail at escorting Brann, it resets correctly
5. Finish the instance and make sure everything is working properly



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
